File: backend/app/models/appointment.py
from sqlalchemy import Column, String
from ..database import Base
import uuid
from datetime import datetime

class Appointment(Base):
    __tablename__ = "appointments"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    client_name = Column(String, nullable=False)
    client_phone = Column(String, nullable=False)
    motorcycle_plate = Column(String, nullable=False)
    motorcycle_model = Column(String, nullable=False)
    service_type = Column(String, nullable=False)  # "mecanica" o "lavado"
    description = Column(String, nullable=False)
    date = Column(String, nullable=False)  # YYYY-MM-DD
    time = Column(String, nullable=False)  # HH:MM
    status = Column(String, nullable=False, default="pendiente")  # "pendiente", "en_progreso", "completado", "cancelado"
    created_at = Column(String, default=lambda: datetime.utcnow().isoformat())

File: frontend/src/types/appointment.ts
export type ServiceType = 'mecanica' | 'lavado';
export type AppointmentStatus = 'pendiente' | 'en_progreso' | 'completado' | 'cancelado';

export interface Appointment {
  id: string;
  client_name: string;
  client_phone: string;
  motorcycle_plate: string;
  motorcycle_model: string;
  service_type: ServiceType;
  description: string;
  date: string;
  time: string;
  status: AppointmentStatus;
  created_at: string;
}

export interface AppointmentCreate {
  client_name: string;
  client_phone: string;
  motorcycle_plate: string;
  motorcycle_model: string;
  service_type: ServiceType;
  description: string;
  date: string;
  time: string;
}

export interface AppointmentUpdate {
  client_name?: string;
  client_phone?: string;
  motorcycle_plate?: string;
  motorcycle_model?: string;
  service_type?: ServiceType;
  description?: string;
  date?: string;
  time?: string;
  status?: AppointmentStatus;
}
